fix(llm-o): replay url builder

nickatrecon opened the pull request:
## Problem

The session replay backlinks in the LLM observability trace views are built with the `urls.replay` base function with undefined replay tabs and filters. The resulting URL (`/replay/home?sessionRecordingId={session_id}`) is problematic because the user's current filters may either filter that session out or bury the session so deep in the session list panel's infinite scroll that it's not selectable without scrolling. Clicking this link often results in the "No recording selected" screen, even though the recording exists.

<!-- Who are we building for, what are their needs, why is this important? -->

<!-- Does this fix an issue? Uncomment the line below with the issue ID to automatically close it when merged -->
<!-- Closes #ISSUE_ID -->

## Changes

~Use the `replaySingle` function instead, which produces a direct-link URL to the specific session (`/replay/{session_id}`). This avoids the filter issues completely.~

Use the `ViewRecordingButton` component instead, which avoids the filter issues completely, and has a first-class modal option! Thanks @lshaowei18 for this idea.

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

## Did you write or update any docs for this change?

<!-- Engineers are responsible for doing the first pass at documenting their features and/or code.  -->

- [ ] I've [added or updated the docs](https://posthog.com/handbook/engineering/writing-docs)
- [ ] I've reached out for help from the docs team
- [x] No docs needed for this change

## How did you test this code?

Tested locally.

From a web app context where session recording is enabled:
1. Generate a UUID for use as a trace ID in steps 2 and 3.
2. [optional] Log several LLM generations to PostHog with manual capture.
3. Log a span with manual capture.
4. The span should show up inside the trace alongside the generations and have a session recording link.

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

## Acknowledgements
@lshaowei18 for multiple tests and improvements.

nickatrecon commented:
> Hi @nickatrecon, I managed to test your code with my localhost and it does work, would you like to mark this as ready for review?
> 
> There is a component called `ViewRecordingButton`, which i'm wondering if it makes sense to use that component since it can open the recording in modal.

Awesome, thank you! I love the suggestion for the ViewRecordingButton instead. I dropped in the change.

How are you testing this? I'll add a snip to the PR and acknowledge you.

lshaowei18 commented:
> > Hi @nickatrecon, I managed to test your code with my localhost and it does work, would you like to mark this as ready for review?
> > There is a component called `ViewRecordingButton`, which i'm wondering if it makes sense to use that component since it can open the recording in modal.
> 
> Awesome, thank you! I love the suggestion for the ViewRecordingButton instead. I dropped in the change.
> 
> How are you testing this? I'll add a snip to the PR and acknowledge you.

Hi @nickatrecon, I tested this locally by:


1) Followed this guide [here](https://posthog.com/tutorials/anthropic-analytics) to have a trace.
2) Visiting this url directly: http://localhost:8010/project/1/llm-observability (I haven't figure out how to turn on feature previews)
3) I haven't figured out how to setup session replay on localhost, so I removed the boolean check `hasSessionID(event)` and replaced the sessionId with a hardcoded one (not ideal but i think it should be sufficient)

lshaowei18 commented on products/llm_observability/frontend/LLMObservabilityTraceScene.tsx:
```diff
@@ -474,15 +475,12 @@ const EventContent = React.memo(({ event }: { event: LLMTrace | LLMTraceEvent |
-                                    </Link>
-                                </div>
+                                <ViewRecordingButton
+                                    fullWidth
```
Can i suggest we remove this and change to `size="xsmall"` instead?

Currently it looks like:
<img width="881" alt="Screenshot 2025-06-18 at 8 43 07 AM" src="https://github.com/user-attachments/assets/02d6fa56-94db-4859-8acb-973531c96b2a" />

With size xsmall:

<img width="901" alt="Screenshot 2025-06-18 at 8 50 45 AM" src="https://github.com/user-attachments/assets/ad47caa7-2a14-4d3a-affd-9ddd6a1c467e" />

lshaowei18 replied:
Hmm im also thinking maybe if we should add a `type='secondary'`, since the other things on the screen mostly has borders:

<img width="638" alt="Screenshot 2025-06-18 at 8 59 18 AM" src="https://github.com/user-attachments/assets/012cd99f-2247-4ef5-8d00-ea908ea9cd7f" />

Let me know what yall think :)

nickatrecon replied:
Love it. Thank you.

lshaowei18 commented on products/llm_observability/frontend/LLMObservabilityTraceScene.tsx:
```diff
@@ -474,15 +475,12 @@ const EventContent = React.memo(({ event }: { event: LLMTrace | LLMTraceEvent |
-                                </div>
+                                <ViewRecordingButton
+                                    fullWidth
+                                    inModal
```
I referred to the other buttons and they seem to add `data-attr`, so maybe we should do that as well?

Something like `data-attr="llm-observability"`?

nickatrecon replied:
Agreed, added it. Thanks!

lshaowei18 reviewed (commented):
Thanks, let me know if you need me to help test anything!

nickatrecon commented:
> 3. I haven't figured out how to setup session replay on localhost, so I removed the boolean check `hasSessionID(event)` and replaced the sessionId with a hardcoded one (not ideal but i think it should be sufficient)

Yup that should work. 

btw: Localhost session replay was problematic for me too. The fix in my case was to run the calling app on a port that isn't used anywhere else in the PostHog stack. The problem was that the PortHog web container uses port 8000 under the hood and when my browser (calling app, which was also using 8000) tries to access recorder.js, it gets a 502 cross-origin error and doesn't load the recorder code. Switching my calling app to run on port 8242 fixed everything. I should probably submit a doc update in the hobby/local dev.

lshaowei18 commented:
> > 3. I haven't figured out how to setup session replay on localhost, so I removed the boolean check `hasSessionID(event)` and replaced the sessionId with a hardcoded one (not ideal but i think it should be sufficient)
> 
> Yup that should work.
> 
> btw: Localhost session replay was problematic for me too. The fix in my case was to run the calling app on a port that isn't used anywhere else in the PostHog stack. The problem was that the PostHog web container uses port 8000 under the hood and when my browser (calling app, which was also using 8000) tried to access recorder.js, it got a 502 cross-origin error and didn't load the recorder code. Switching my calling app to run on port 8242 fixed everything. I should probably submit a doc update in the hobby/local dev.

Nice! Yes please that will be great, maybe you can update the docs [here](https://posthog.com/handbook/engineering/developing-locally)?

This is the github repo :) https://github.com/PostHog/posthog.com
